### PR TITLE
CI: Autopublish using GitHub Actions + spec-prod

### PR DESCRIPTION
**Pull request by sideshowbarker**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
do we still need the make file?

File: .github/workflows/autopublish.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  main:
+    name: Build, Validate, and Publish
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: w3c/spec-prod@v2
+        with:
+          GH_PAGES_BRANCH: gh-pages
```
Comment (marcoscaceres) [inline .github/workflows/autopublish.yml:15]:
```suggestion
          TOOLCHAIN: bikeshed
          GH_PAGES_BRANCH: gh-pages
          SOURCE: index.src.html
```

**Comment by sideshowbarker** [comment]
> do we still need the make file?

I guess not — now deleted

File: .github/workflows/autopublish.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v2
+      - uses: w3c/spec-prod@v2
+        with:
+          TOOLCHAIN: bikeshed
+          GH_PAGES_BRANCH: gh-pages
+          SOURCE: index.src.html
+          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
+          W3C_WG_DECISION_URL: https://lists.w3.org/Archives/Public/public-webappsec/2015Mar/0170.html
+          W3C_BUILD_OVERRIDE: |
+            shortname: secure-contexts
```
Comment (marcoscaceres) [inline .github/workflows/autopublish.yml:21]:
```suggestion
            shortname: secure-contexts
            specStatus: WD
```

File: .github/workflows/autopublish.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: w3c/spec-prod@v2
+        with:
+          TOOLCHAIN: bikeshed
+          GH_PAGES_BRANCH: gh-pages
+          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
+          W3C_WG_DECISION_URL: https://lists.w3.org/Archives/Public/public-webappsec/2015Mar/0170.html
+          W3C_BUILD_OVERRIDE: |
+            shortname: secure-contexts
```
Comment (marcoscaceres) [inline .github/workflows/autopublish.yml:20]: Oops, I think we want "CRD", right? 

```suggestion
            shortname: secure-contexts
            specStatus: CRD
```

File: .github/workflows/autopublish.yml (hunk 1 above)
Comment (sideshowbarker) [reply .github/workflows/autopublish.yml:16]: If we’re not using a non-default SOURCE value, do we actually need TOOLCHAIN?

File: .github/workflows/autopublish.yml (hunk 3 above)
Comment (sideshowbarker) [reply .github/workflows/autopublish.yml:20]: > Oops, I think we want "CRD", right?

Yeah

**Review by marcoscaceres — APPROVED** [review]
Looking good! 🤞

**Comment by marcoscaceres** [comment]
Woot! the system works. Bikeshed is upset.

**Comment by marcoscaceres** [comment]
@sideshowbarker, Bikeshed is complaining for the right reasons:

```
   LINE ~514: Multiple possible 'environment' dfn refs.
680
      Arbitrarily chose https://html.spec.whatwg.org/multipage/infrastructure.html#blob-url-entry-environment
681
      The following refs show up multiple times in their spec, in a way that Bikeshed can't distinguish between. Either create a manual link, or ask the spec maintainer to add disambiguating attributes (usually a for='' attribute to all of them).
682
      spec:html; type:dfn; for:/; text:environment
683
        https://html.spec.whatwg.org/multipage/infrastructure.html#blob-url-entry-environment
684
        https://html.spec.whatwg.org/multipage/webappapis.html#environment
685
      LINE ~518: Multiple possible 'environment' dfn refs.
686
      Arbitrarily chose https://html.spec.whatwg.org/multipage/infrastructure.html#blob-url-entry-environment
687
      The following refs show up multiple times in their spec, in a way that Bikeshed can't distinguish between. Either create a manual link, or ask the spec maintainer to add disambiguating attributes (usually a for='' attribute to all of them).
688
      spec:html; type:dfn; for:/; text:environment
689
        https://html.spec.whatwg.org/multipage/infrastructure.html#blob-url-entry-environment
690
        https://html.spec.whatwg.org/multipage/webappapis.html#environment
691
       ✔  Successfully generated, with 2 linking errors
692
    [INFO] Finding relevant assets…
693
      Local assets to be copied: []
694
      Remote assets to be downloaded: []
695
    Copying 0 local files…
696
    Downloading 0 remote files…
697
    $ ls -R
698
      .:
699
      index.html
700
  [INFO] Build bikeshed document "index.bs" (w3c)…
701
    $ bikeshed --die-on=fatal spec "index.bs" "index.bs.built.html" --md-shortname="secure-contexts" --md-specStatus="CRD" --md-date="2021-09-07" --md-previous-version="https://www.w3.org/TR/2016/CR-secure-contexts-20160915/"
702
      FATAL ERROR: Unknown metadata key "Specstatus". Prefix custom keys with "!".
703
       ✘  Did not generate, due to fatal errors
704
    Failed
705
  Error: Process completed with exit code 2.
```

**Comment by marcoscaceres** [comment]
FATAL ERROR: Unknown metadata key "Specstatus". Prefix custom keys with "!".

**Comment by sideshowbarker** [comment]
> FATAL ERROR: Unknown metadata key "Specstatus". Prefix custom keys with "!".

Yeah I don’t understand that error — that `Specstatus` is something that spec-prod added, as far as I can see

**Comment by marcoscaceres** [comment]
Ah, maybe it should "Status" over on the yaml file... my bad!

File: .github/workflows/autopublish.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: w3c/spec-prod@v2
+        with:
+          GH_PAGES_BRANCH: gh-pages
+          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
+          W3C_WG_DECISION_URL: https://lists.w3.org/Archives/Public/public-webappsec/2015Mar/0170.html
+          W3C_BUILD_OVERRIDE: |
+            shortname: secure-contexts
+            specStatus: CRD
```
Comment (marcoscaceres) [inline .github/workflows/autopublish.yml:20]:
```suggestion
            Status: CRD
```

**Comment by sideshowbarker** [comment]
> Ah, maybe it should "Status" over on the yaml file... my bad!

no worries — easy to fix :)

**Comment by marcoscaceres** [comment]
WDY we use https://wpt.fyi/results/secure-contexts?label=experimental&label=master&aligned for the implementation report?

**Comment by marcoscaceres** [comment]
Or just: https://wpt.fyi/results/secure-contexts

**Comment by sideshowbarker** [comment]
> Or just: [wpt.fyi/results/secure-contexts](https://wpt.fyi/results/secure-contexts)

Yup, let’s go with that

Also the Issue Tracking addition

You gonna make the updates? Or should I?

**Comment by marcoscaceres** [comment]
Could you please do it. I haven't checkout out the repo 🥴

**Comment by marcoscaceres** [comment]
(I'll back-seat-driver make comments tho! it will be awesome)

**Comment by sideshowbarker** [comment]
> Could you please do it. I haven't checkout out the repo 🥴

Ok, yeah, I will after a bit. I'm on from my phone right now

**Comment by sideshowbarker** [comment]
After merging, a bunch more errors for stuff that as far as I can see, Bikeshed/spec-prod should be adding automatically, based on the Status value.

https://github.com/w3c/webappsec-secure-contexts/runs/3531636253#step:3:670

**Pull request merged**